### PR TITLE
Allow listening for options changes

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

in this PR I am adding two convenience extensions that can be used when using the `AddResilienceStrategy` extension method:

- `EnableReloads<TOptions>`: enables the dynamic reloads of resilience strategy whenever `TOptions` are changed. 
- `GetOptions<TOptions>`: shorthand for calling `context.ServiceProvider.GetRequiredService<IOptionsMonitor<TOptions>>().Get(name)`

These should help with the most common scenarios when defining the resilience strategy using the DI container. 

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -7,8 +7,10 @@
     <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="1.0.0" />
     <PackageVersion Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.1" />
     <PackageVersion Include="Microsoft.Extensions.Caching.Memory" Version="7.0.0" />
+    <PackageVersion Include="Microsoft.Extensions.Configuration" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Logging" Version="7.0.0" />
+    <PackageVersion Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="7.0.0" />
```
Comment (martincostello) [inline src/Directory.Packages.props:13]: Let's add both of these to the dependabot exclusions:

https://github.com/App-vNext/Polly/blob/90c9894ced5bf781bc5229abf377f58206cee42b/.github/dependabot.yml#L18-L27

File: src/Polly.Extensions.Tests/ReloadableResilienceStrategyTests.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Moq;
+using Polly.Extensions.Utils;
+using Polly.Registry;
+
+namespace Polly.Extensions.Tests;
```
Comment (martincostello) [inline src/Polly.Extensions.Tests/ReloadableResilienceStrategyTests.cs:10]:
```suggestion
namespace Polly.Extensions.Tests;

```

File: src/Polly.Extensions.Tests/ReloadableResilienceStrategyTests.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Threading.Tasks;
```
Comment (martincostello) [inline src/Polly.Extensions.Tests/ReloadableResilienceStrategyTests.cs:2]: These two are redundant.

File: src/Polly.Extensions/DependencyInjection/AddResilienceStrategyContext.cs [hunk 4]
```diff
@@ -30,4 +33,40 @@ internal AddResilienceStrategyContext(ConfigureBuilderContext<TKey> registryCont
...
+        var helper = registry.Get<TOptions>(StrategyKey, name);
+
+        RegistryContext.EnableReloads(helper.GetCancellationToken);
+    }
+
+    /// <summary>
+    /// Gets the options identified by <paramref name="name"/>.
+    /// </summary>
+    /// <typeparam name="TOptions">The options type.</typeparam>
+    /// <param name="name">Tje options name, if any.</param>
```
Comment (martincostello) [inline src/Polly.Extensions/DependencyInjection/AddResilienceStrategyContext.cs:61]:
```suggestion
    /// <param name="name">The options name, if any.</param>
```

File: src/Polly.Extensions/Utils/OptionsReloadHelper.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
+using System;
+using Microsoft.Extensions.Options;
+
+namespace Polly.Extensions.Utils;
+
+internal class OptionsReloadHelper<T> : IDisposable
```
Comment (martincostello) [inline src/Polly.Extensions/Utils/OptionsReloadHelper.cs:6]:
```suggestion
internal sealed class OptionsReloadHelper<T> : IDisposable
```

File: src/Polly.Extensions/Utils/OptionsReloadHelper.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
+using System;
```
Comment (martincostello) [inline src/Polly.Extensions/Utils/OptionsReloadHelper.cs:1]:
```suggestion
```

File: src/Polly.Extensions/Utils/OptionsReloadHelperRegistry.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
+using System;
```
Comment (martincostello) [inline src/Polly.Extensions/Utils/OptionsReloadHelperRegistry.cs:1]:
```suggestion
```

File: src/Polly.Extensions/Utils/OptionsReloadHelperRegistry.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Polly.Extensions.Utils;
+
+internal class OptionsReloadHelperRegistry<TKey> : IDisposable
```
Comment (martincostello) [inline src/Polly.Extensions/Utils/OptionsReloadHelperRegistry.cs:7]:
```suggestion
internal sealed class OptionsReloadHelperRegistry<TKey> : IDisposable
```

File: src/Directory.Packages.props (hunk 1 above)
Comment (martintmk) [reply src/Directory.Packages.props:13]: Even though it's only used in tests? I think these can be automatically updated.

**Review by martincostello — APPROVED** [review]

**Pull request merged**